tests/test_deprecated.py

```diff
@@ -0,0 +1,41 @@
+"""Unittests for deprecated features"""
+
+
+import affine
+import pytest
+
+import rasterio
+from rasterio.transform import guard_transform
+
+
+def test_open_affine_and_transform(path_rgb_byte_tif):
+    with pytest.raises(ValueError):
```

Comment:
@geowurster this may be overkill, but since lots of things can raise `ValueError` and `TypeError`, it may be a good idea to check the specific exception text like [this](https://github.com/mapbox/rasterio/blob/master/tests/test_write.py#L282) just to make sure you got the one you expected.